src/components/DistanceEReceipt.js

```diff
@@ -0,0 +1,121 @@
+import React, {useMemo} from 'react';
+import {View, ScrollView} from 'react-native';
+import lodashGet from 'lodash/get';
+import _ from 'underscore';
+import Text from './Text';
+import styles from '../styles/styles';
+import transactionPropTypes from './transactionPropTypes';
+import * as ReceiptUtils from '../libs/ReceiptUtils';
+import * as ReportUtils from '../libs/ReportUtils';
+import * as CurrencyUtils from '../libs/CurrencyUtils';
+import * as TransactionUtils from '../libs/TransactionUtils';
+import tryResolveUrlFromApiRoot from '../libs/tryResolveUrlFromApiRoot';
+import ThumbnailImage from './ThumbnailImage';
+import useLocalize from '../hooks/useLocalize';
+import Icon from './Icon';
+import themeColors from '../styles/themes/default';
+import * as Expensicons from './Icon/Expensicons';
+import EReceiptBackground from '../../assets/images/eReceipt_background.svg';
+import useNetwork from '../hooks/useNetwork';
+import PendingMapView from './MapView/PendingMapView';
+
+const propTypes = {
+    /** The transaction for the distance request */
+    transaction: transactionPropTypes,
+};
+
+const defaultProps = {
+    transaction: {},
+};
+
+function DistanceEReceipt({transaction}) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const {thumbnail} = ReceiptUtils.getThumbnailAndImageURIs(transaction.receipt.source, transaction.filename);
```

Comment:
I have a little worry about this. In my understanding, the `transaction` is optional, but the component will break if we don't pass the `transaction` param.

We can keep the `transaction` as is, and to prevent the component break, we need to check if the transaction has a receipt.

```suggestion
    const {thumbnail} = TransactionUtils.hasReceipt(transaction) ? ReceiptUtils.getThumbnailAndImageURIs(transaction.receipt.source, transaction.filename) : {};
```

![Screenshot_1696529250](https://github.com/Expensify/App/assets/25520267/6453804d-066f-4db1-aa9d-46cda823c37d)

Reply:
Oh yeah good point! That looks wrong. I'll get it cleaned up soon.